### PR TITLE
Move fluentd es images

### PR DESCRIPTION
**Pull request by coffeepac**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
moves addon image to community controlled repository
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Move fluentd-elasticsearch addon images to community controlled location
```

### CODE REVIEW

**Comment by monotek** [comment]
Could you explain what the whole go magic is for in the es image directory?
I'm used to te elasticsearch helm chart from https://github.com/helm/charts/tree/master/stable/elasticsearch where no extras are needed for es node discovery.
So is this realy needed?

As this is an example we also might consider to run Elasticsearch with only one node to save resources in test setups? This would work via some additional env vars:
```
EXPECTED_MASTER_NODES: "1"
MINIMUM_MASTER_NODES: "1"
RECOVER_AFTER_MASTER_NODES: "1"
```



We could also use this pr to update the Elasticsearch and Kibana apps to the most recent version: 6.6.0
Thoughts?

**Review by monotek — COMMENTED** [review]
Could you explain what the whole go magic is for in the es image directory?
I'm used to te elasticsearch helm chart from https://github.com/helm/charts/tree/master/stable/elasticsearch where no extras are needed for es node discovery.
So is this realy needed?

As this is an example we also might consider to run Elasticsearch with only one node to save resources in test setups? This would work via some additional env vars:

```
EXPECTED_MASTER_NODES: "1"
MINIMUM_MASTER_NODES: "1"
RECOVER_AFTER_MASTER_NODES: "1"
```
We could also use this pr to update the Elasticsearch and Kibana apps to the most recent version: 6.6.0
Thoughts?

**Comment by desaintmartin** [comment]
If this is an example, should we point to "production-grade" solutions like the community-maintained Helm Charts for this?

**Comment by monotek** [comment]
I like the idea but we would introduce the Helm dependency with that. 

Elastic now provides own charts (https://github.com/elastic/helm-charts) which are also already listed on helmhub (https://hub.helm.sh/charts/elastic) but i have not used them until now.

Not sure which ones we should prefer.
Maybe link to both?

**Comment by desaintmartin** [comment]
Because the "official" Helm Chart is still alpha, I would link to the community-maintained one(s) like charts.git/elasticstack.

Helm is only an example, it would be interesting for someone looking around to know what the different opportunities he has in order to install a proper EFK stack.
If referencing Helm is out of scope, this list may be for information purpose only.

Anyway, It is really annoying from a newcomer a/ to have to install everything by himself, only to realize that b/ it does not work on production and he has to find how to do it.

File: cluster/addons/fluentd-elasticsearch/es-image/Dockerfile [hunk 1]
```diff
@@ -12,12 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+FROM golang:1.11.4 AS builder
```
Comment (ialidzhikov) [inline cluster/addons/fluentd-elasticsearch/es-image/Dockerfile]: Is it possible to use `1.11.5`?

**Comment by monotek** [comment]
> If this is an example, should we point to "production-grade" solutions like the community-maintained Helm Charts for this?

@desaintmartin 
I've added another PR with such note: https://github.com/kubernetes/kubernetes/pull/74209

File: cluster/addons/fluentd-elasticsearch/es-image/Dockerfile (hunk 1 above)
Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/es-image/Dockerfile]: yes.

**Comment by coffeepac** [comment]
the go application is used to populate the node list.  it looks like that application can be removed at this point.  However, I would like to do that in a different PR to limit the scope of this one.

**Comment by brancz** [comment]
We've talked about this a number of times, and moving the images to a community controlled repository has my :+1: until google finds a way for non googlers to publish images. We've been promised the ability to do this since the Kubernetes 1.6 (!!!) release, and this still hasn't happened. Because of that, this has my full support.

**Comment by coffeepac** [comment]
@fejta or @spiffxp can you take a look at the change in `hack/`?  Its adding `go.sum` to the files that shouldn't be spellchecked.  There's a go binary in this image.

Tagging both of you as this OWNERS file is, maybe, the busiest people in the project and both of you commented on a different recent PR in this addon.

File: hack/.spelling_failures [hunk 2]
```diff
@@ -7,6 +7,7 @@ docs/man
 docs/user-guide
 generated.pb.go
 generated.proto
+go.sum
```
Comment (coffeepac) [inline hack/.spelling_failures]: should I also add `go.mod` here?  Its not currently failing the check but it easily could due to github handles.

Comment (spiffxp) [reply hack/.spelling_failures]: sounds like a good idea to me

**Comment by spiffxp** [comment]
/approve
/hold
feel free to remove the /hold when you've added `go.mod` and addressed the `1.11.5` change
/priority important-soon

File: hack/.spelling_failures (hunk 2 above)
Comment (coffeepac) [reply hack/.spelling_failures]: added.

File: cluster/addons/fluentd-elasticsearch/es-image/Dockerfile (hunk 1 above)
Comment (coffeepac) [reply cluster/addons/fluentd-elasticsearch/es-image/Dockerfile]: updated.

**Pull request merged**